pkg/core/fs_ro_ops.go

```diff
@@ -24,35 +24,130 @@ func (fs *readOnlyFsInternal) StatFS(
 	return statFS()
 }
 
-func (fs *readOnlyFsInternal) LookUpInode(ctx context.Context, op *fuseops.LookUpInodeOp) error {
-	log.Print(fmt.Printf("lookup parent id:%d, child: %s ", op.Parent, op.Name))
+func (fs *readOnlyFsInternal) opStart(op interface{}) {
+	opType := reflect.TypeOf(op)
+	switch op.(type) {
+	case *fuseops.ReadFileOp:
+		r := op.(*fuseops.ReadFileOp)
+		fs.l.Info("Start",
+			zap.String("Request", opType.String()),
+			zap.String("repo", fs.bundle.RepoID),
+			zap.String("bundle", fs.bundle.BundleID),
+			zap.Uint64("inode", uint64(r.Inode)),
+		)
+		return
+	case *fuseops.WriteFileOp:
+		w := op.(*fuseops.WriteFileOp)
+		fs.l.Info("Start",
+			zap.String("Request", opType.String()),
+			zap.String("repo", fs.bundle.RepoID),
+			zap.String("bundle", fs.bundle.BundleID),
+			zap.Uint64("inode", uint64(w.Inode)),
+		)
+		return
+	case *fuseops.ReadDirOp:
+		r := op.(*fuseops.ReadDirOp)
+		fs.l.Info("Start",
+			zap.String("Request", opType.String()),
+			zap.String("repo", fs.bundle.RepoID),
+			zap.String("bundle", fs.bundle.BundleID),
+			zap.Uint64("inode", uint64(r.Inode)),
+		)
+		return
+	}
+	fs.l.Info("Start",
+		zap.String("Request", opType.String()),
+		zap.String("repo", fs.bundle.RepoID),
+		zap.String("bundle", fs.bundle.BundleID),
+		zap.Any("op", op),
+	)
+}
+func (fs *readOnlyFsInternal) opEnd(op interface{}, err error) {
+	opType := reflect.TypeOf(op)
+	switch op.(type) {
```

Comment:
i suspect it's more idiomatic to assign a value at the top of the `switch` with, for example, `v := op.(type)` rather than adding duplicate type assertions in each `case`.

Reply:
Thank you!